shared/lib/sites/test/reducers.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SET_SELECTED_SITE,
+	RECEIVE_SITE
+} from '../action-types';
+import {
+	selected,
+	byId
+} from '../reducers';
+
+describe( 'reducers', () => {
+	describe( '#selected()', () => {
+		it( 'should default to null', () => {
+			const state = selected( undefined, {} );
+
+			expect( state ).to.be.null;
+		} );
+
+		it( 'should set the selected site ID', () => {
+			const state = selected( null, {
+				type: SET_SELECTED_SITE,
+				siteId: 2916284
+			} );
+
+			expect( state ).to.equal( 2916284 );
+		} );
+	} );
+
+	describe( '#byId()', () => {
+		it( 'should default to an empty object', () => {
+			const state = byId( undefined, {} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should index sites by ID', () => {
+			const state = byId( null, {
+				type: RECEIVE_SITE,
+				site: { ID: 2916284, name: 'WordPress.com Example Blog' }
+			} );
+
+			expect( state ).to.eql( {
+				2916284: { ID: 2916284, name: 'WordPress.com Example Blog' }
+			} );
+		} );
+
```

Comment:
might want a test to prove that the byId map becomes a new instance when a site is added